lib/torrent.js

```diff
@@ -395,23 +363,96 @@ Torrent.prototype._onListening = function () {
     self.emit('warning', err)
   }
 
-  // if full metadata was included in initial torrent id, use it immediately. Otherwise,
-  // wait for torrent-discovery to find peers and ut_metadata to get the metadata.
-  if (self.info) self._onMetadata(self)
+  if (self.info) {
+    // if full metadata was included in initial torrent id, use it immediately. Otherwise,
+    // wait for torrent-discovery to find peers and ut_metadata to get the metadata.
+    self._onMetadata(self)
+  } else if (self.xs) {
+    self._getMetadataFromServer()
```

Comment:
Moved to after listening event since it's really fast and it's when the other discovery happens. This may prevent future bugs where we expect to already be listening.